projects/helper/chain/radixdlt.js

```diff
@@ -37,12 +40,57 @@ async function queryAddresses({ addresses = [], }) {
   return items
 }
 
+async function queryLsus(addresses = []) {
+  // Receives a list of lsu addresses and returns the redemption value in XRD
+
+  let lsuRedemptionValues = {}
+  const chunks  = sliceIntoChunks(addresses, 20)
+  for (const chunk of chunks) {
+    let body = {
+      "addresses": chunk,
+      "aggregation_level": "Vault"
+    }
+    let data = await post(ENTITY_DETAILS_URL, body)
+
+    let validators = []
+    for (const lsuResource of data.items) {
+      let validator = lsuResource.metadata.items.filter(metadataItem => metadataItem.key === "validator")[0]
+      lsuRedemptionValues[lsuResource.address] = {
+        "totalSupplyOfStakeUnits": lsuResource.details.total_supply,
+        "validatorAddress": validator.value.typed.value
+      }
+
+      validators.push(validator.value.typed.value)
+    }
+
+    let validatorsDetailsBody = {
+      "addresses": validators,
+      "aggregation_level": "Vault"
+    }
+    let validatorsData = await post(ENTITY_DETAILS_URL, validatorsDetailsBody)
+    for (const validator of validatorsData.items) {
+      let stakeUnitResourceAddress = validator.details.state.stake_unit_resource_address
+      let stakeXrdVaultAddress = validator.details.state.stake_xrd_vault.entity_address
+
+      let xrdStakeResource = validator.fungible_resources.items.filter(item => item.resource_address === XRD_RESOURCE_ADDRESS)
+      let xrdStakeVault = xrdStakeResource[0].vaults.items.filter(vault => vault.vault_address === stakeXrdVaultAddress)
+      let xrdStakeVaultBalance = new BigNumber(xrdStakeVault[0].amount)
```

Comment:
Oh dang, this should have been easier, but I think you're right doing it this way - the Gateway doesn't properly return this as a Gateway model from state details - only from the validator set endpoint.

Reply:
🎯